src/test/java/edu/ucsb/cs156/example/controllers/PersonalSchedulesControllerTests.java

```diff
@@ -0,0 +1,565 @@
+
+        // arrange
+
+        User thisUser = currentUserService.getCurrentUser().getUser();
+
+        PersonalSchedule p1 = PersonalSchedule.builder().Name("Name 1").Description("Description 1").Quarter("Quarter 1").user(thisUser).id(1L).build();
+        PersonalSchedule p2 = PersonalSchedule.builder().Name("Name 2").Description("Description 2").Quarter("Quarter 2").user(thisUser).id(2L).build();
+
+        ArrayList<PersonalSchedule> expectedSchedules = new ArrayList<>();
+        expectedSchedules.addAll(Arrays.asList(p1, p2));
+        when(personalscheduleRepository.findAllByUserId(thisUser.getId())).thenReturn(expectedSchedules);
+
+        // act
+        MvcResult response = mockMvc.perform(get("/api/personalschedules/all"))
+                .andExpect(status().isOk()).andReturn();
+
+        // assert
+
+        verify(personalscheduleRepository, times(1)).findAllByUserId(eq(thisUser.getId()));
+        String expectedJson = mapper.writeValueAsString(expectedSchedules);
+        String responseString = response.getResponse().getContentAsString();
+        assertEquals(expectedJson, responseString);
+    }
+
+    @WithMockUser(roles = { "USER" })
+    @Test
+    public void api_schedules_post__user_logged_in() throws Exception {
+        // arrange
+
+        User u = currentUserService.getCurrentUser().getUser();
+
+        PersonalSchedule expectedSchedule = PersonalSchedule.builder()
+                .Name("Test Name")
+                .Description("Test Description")
+                .Quarter("Test Quarter")
+                .user(u)
+                .id(0L)
+                .build();
+
+        when(personalscheduleRepository.save(eq(expectedSchedule))).thenReturn(expectedSchedule);
+
+        // act
+        MvcResult response = mockMvc.perform(
+                post("/api/personalschedules/post?Name=Test Name&Description=Test Description&Quarter=Test Quarter")
+                        .with(csrf()))
+                .andExpect(status().isOk()).andReturn();
+
+        // assert
+        verify(personalscheduleRepository, times(1)).save(expectedSchedule);
+        String expectedJson = mapper.writeValueAsString(expectedSchedule);
+        String responseString = response.getResponse().getContentAsString();
+        assertEquals(expectedJson, responseString);
+    }
+
+    // @WithMockUser(roles = { "USER" })
+    // @Test
+    // public void api_schedules__user_logged_in__delete_todo() throws Exception {
+    //     // arrange
+
+    //     User u = currentUserService.getCurrentUser().getUser();
+    //     Todo todo1 = Todo.builder().title("Todo 1").details("Todo 1").done(false).user(u).id(15L).build();
+    //     when(todoRepository.findByIdAndUser(eq(15L), eq(u))).thenReturn(Optional.of(todo1));
+
+    //     // act
+    //     MvcResult response = mockMvc.perform(
+    //             delete("/api/todos?id=15")
+    //                     .with(csrf()))
```

Comment:
Why is this commented out code here?   It should probably be deleted.See today's lecture notes (the section about commented out code)

https://ucsb-cs156.github.io/w22/lectures/lect26/